pkg/target/target_test.go

```diff
@@ -104,3 +104,194 @@ func TestProcessLabelValues(t *testing.T) {
+	if err != nil {
+		t.Fatalf("error during label processing %v", err)
+	}
+
+	clusterName, ok := templatedValues["clusterName"]
+	if !ok {
+		t.Fatal("key clusterName not found")
+	}
+
+	if clusterName != "local" {
+		t.Fatal("unable to assert correct clusterName")
+	}
+
+	fromAnnotation, ok := templatedValues["fromAnnotation"]
+	if !ok {
+		t.Fatal("key fromAnnotation not found")
+	}
+
+	if fromAnnotation != "test" {
+		t.Fatal("unable to assert correct value for fromAnnotation")
+	}
+
+	clusterNamespace, ok := templatedValues["clusterNamespace"]
+	if !ok {
+		t.Fatal("key clusterNamespace not found")
+	}
+
+	if clusterNamespace != "dev-clusters" {
+		t.Fatal("unable to assert correct value for clusterNamespace")
+	}
+
+	fleetClusterName, ok := templatedValues["fleetClusterName"]
+	if !ok {
+		t.Fatal("key clusterName not found")
+	}
+
+	if fleetClusterName != "my-cluster" {
+		t.Fatal("unable to assert correct value fleetClusterName")
+	}
+
+	reallyLongClusterName, ok := templatedValues["reallyLongClusterName"]
+	if !ok {
+		t.Fatal("key reallyLongClusterName not found")
+	}
+
+	if reallyLongClusterName != "kubernets.io/cluster/foobar" {
+		t.Fatal("unable to assert correct value reallyLongClusterName")
+	}
+
+	customStruct, ok := templatedValues["customStruct"].([]interface{})
+	if !ok {
+		t.Fatal("key customStruct not found")
+	}
+
+	firstMap, ok := customStruct[0].(map[string]interface{})
+	if !ok {
+		t.Fatal("unable to assert first element to map[string]interface{}")
+	}
+
+	firstElemVal, ok := firstMap["name"]
+	if !ok {
+		t.Fatal("unable to find key name in the first element of customStruct")
+	}
+
+	if firstElemVal.(string) != "foo" {
+		t.Fatal("label replacement not performed in first element")
+	}
+
+	secondElement, ok := customStruct[1].(map[string]interface{})
+	if !ok {
+		t.Fatal("unable to assert second element of customStruct to map[string]interface{}")
+	}
+
+	secondElemVal, ok := secondElement["element2"]
+	if !ok {
+		t.Fatal("unable to find key element2")
+	}
+
+	if secondElemVal.(string) != "bar" {
+		t.Fatal("template replacement not performed in second element")
+	}
+
+	thirdElement, ok := customStruct[2].(map[string]interface{})
+	if !ok {
+		t.Fatal("unable to assert second element of customStruct to map[string]interface{}")
+	}
+
+	thirdElemVal, ok := thirdElement["element3_dev"]
+	if !ok {
+		t.Fatal("unable to find key element3_dev")
+	}
+
+	if thirdElemVal.(string) != "local" {
+		t.Fatal("template replacement not performed in third element")
+	}
+
+	funcs, ok := templatedValues["funcs"].(map[string]interface{})
+	if !ok {
+		t.Fatal("key funcs not found")
+	}
+
+	upper, ok := funcs["upper"]
+	if !ok {
+		t.Fatal("key upper not found")
+	}
+
+	if upper.(string) != "FOO_test" {
+		t.Fatal("upper func was not right")
+	}
+
+	join, ok := funcs["join"]
+	if !ok {
+		t.Fatal("key join not found")
+	}
+
+	if join.(string) != "alpha,beta,omega" {
+		t.Fatal("join func was not right")
+	}
+
+}
```

Comment:
Can we also add a test to demonstrate the usage of Helm.DisablePreProcess: true usecase?

Reply:
Added.